File: examples-sdk/go/001/main.go
// 001 — AI Gateway (SSE Pipeline)
// Equivalent to: examples/001-basic-ai-gw-demo (Rust)
// Compile: vil compile --from go --input 001/main.go --release
package main

import (
	"fmt"
	"os"

	vil "github.com/OceanOS-id/vil-go"
)

func main() {
	pipeline := vil.NewPipeline("ai-gateway", 3080)

	// -- Semantic types -------------------------------------------------------
	pipeline.SemanticType("InferenceState", "state", map[string]string{
		"request_id": "u64",
		"tokens":     "u32",
	})
	pipeline.Fault("InferenceFault", []string{"UpstreamTimeout", "ParseError"})

	// -- Nodes ----------------------------------------------------------------
	pipeline.Sink(vil.SinkOpts{Port: 3080, Path: "/trigger", Name: "webhook"})
	pipeline.Source(vil.SourceOpts{
		URL:     "http://localhost:4545/v1/chat/completions",
		Format:  "sse",
		Dialect: "openai",
		JsonTap: "choices[0].delta.content",
		Name:    "inference",
	})

	// -- Tri-Lane routes ------------------------------------------------------
	pipeline.Route("webhook.trigger_out", "inference.trigger_in", "LoanWrite")
	pipeline.Route("inference.data_out", "webhook.data_in", "LoanWrite")
	pipeline.Route("inference.ctrl_out", "webhook.ctrl_in", "Copy")

	// -- Emit / compile -------------------------------------------------------
	if os.Getenv("VIL_COMPILE_MODE") == "manifest" {
		fmt.Println(pipeline.ToYAML())
	} else {
		pipeline.Compile()
	}
}

File: examples-sdk/zig/001/main.zig
// 001 — AI Gateway (SSE Pipeline)
// Equivalent to: examples/001-basic-ai-gw-demo (Rust)
// Compile: vil compile --from zig --input 001/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var pipeline = vil.VilPipeline.init("ai-gateway", 3080);

    // Nodes
    pipeline.sink("webhook", 3080, "/trigger");
    pipeline.source("inference", "http://localhost:4545/v1/chat/completions", .{
        .format = "sse",
        .dialect = "openai",
        .json_tap = "choices[0].delta.content",
    });

    // Tri-Lane routes
    pipeline.route("webhook.trigger_out", "inference.trigger_in", "LoanWrite");
    pipeline.route("inference.data_out", "webhook.data_in", "LoanWrite");
    pipeline.route("inference.ctrl_out", "webhook.ctrl_in", "Copy");

    pipeline.compile();
}
